src/Sarif.Viewer.VisualStudio.Core/Services/DataService.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using EnvDTE;
+
+using EnvDTE80;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.Sarif.Viewer.ErrorList;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Sarif.Viewer.Services
+{
+    /// <inheritdoc/>
+    public class DataService : SDataService, IDataService
+    {
+        /// <inheritdoc/>
+        public void SendEnhancedResultData(SarifLog sarifLog)
+        {
+            SendEnhancedResultDataAsync(sarifLog).FileAndForget(Constants.FileAndForgetFaultEventNames.SendEnhancedData);
+        }
+
+        private static async Task SendEnhancedResultDataAsync(SarifLog sarifLog)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            Assumes.NotNull(sarifLog);
+            Assumes.True(sarifLog.Runs?.Count == 1);
+
+            var componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
+            if (componentModel != null)
+            {
+                var dte = Package.GetGlobalService(typeof(DTE)) as DTE2;
+                var projectNameCache = new ProjectNameCache(dte?.Solution);
+
+                Result result = sarifLog.Runs.First().Results.First();
+                var sarifErrorListItem = new SarifErrorListItem(sarifLog.Runs.First(), 0, result, string.Empty, projectNameCache);
```

Comment:
>sarifErrorListItem [](http://example.com/codeflow?start=20&length=18)

wondering if directly create SarifErrorListItem should work or not. It needs CodeAnalysisResultManager to pre-process results and create cache etc. See ErrorListService.WriteRunToErrorList() for example.

Another option maybe is re-use ErrorListService.ProcessSarifLogAsync() if possible.

Reply:
Yeah I think we'll need to do some refactoring to accommodate this scenario